### PR TITLE
TM-81 Persistence Service

### PR DESCRIPTION
**Pull request by dan0bar**
Connection to DB and HealthCheck

### CODE REVIEW

File: be/src/services/persistence-service.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
+"use strict";
```
Comment (ManuelTS) [inline be/src/services/persistence-service.js]: Please don't commit .js files, only the .ts ones.

File: be/src/controlers/health.router.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import {Request, Response, Router} from 'express';
+import PersistenceService from "../services/persistence-service";
+
+const router: Router = Router();
+
+router.get('/health', async (req: Request, res: Response) => {
+    console.log("HEALTH")
```
Comment (ManuelTS) [inline be/src/controlers/health.router.ts]: As we get a response either way on calling `/health`, I think this logging can be removed

File: be/src/controlers/health.router.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    const health_check = await PersistenceService.pingDB()
+
+    if (health_check) {
+        res.json({statusCode: 200, greeting: "Hello World"});
```
Comment (ManuelTS) [inline be/src/controlers/health.router.ts]: Why returning `Hello World` in a ping response?
```suggestion
        res.status(200);
```

File: be/src/controlers/health.router.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    } else {
+        res.status(500).json({statusCode: 500, greeting: "Hello World"});
```
Comment (ManuelTS) [inline be/src/controlers/health.router.ts]:
```suggestion
        res.json({statusCode: 500, message: "Ping to mongoDB failed"});
```

File: be/src/services/persistence-service.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import {Db, MongoClient} from 'mongodb';
+
+class PersistenceService {
+    private client: MongoClient;
+    private db: Db;
+    private MONGO_DATABASE: string
+
+
+    constructor() {
+        console.log("----------------PersistenceService")
```
Comment (ManuelTS) [inline be/src/services/persistence-service.ts]: This log seams unnecessary

File: be/src/services/persistence-service.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    create(): void {
+        console.log("create");
+    }
+}
+
+export default new PersistenceService();
+
+// interface Entity {
```
Comment (ManuelTS) [inline be/src/services/persistence-service.ts]: Don't commit commented code.

File: be/src/services/entity.model.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+let Entity = {
```
Comment (ManuelTS) [inline be/src/services/entity.model.ts]: It seams you are not using this empty Entity object anywhere, why not deleting it?

File: be/src/controlers/health.router.ts (hunk 3 above)
Comment (ManuelTS) [reply be/src/controlers/health.router.ts]: @ja-fra is returing 200 sufficient? Is anything of
https://docs.mongodb.com/v4.2/reference/command/serverStatus/
needed, in another method maybe?

**Review by ManuelTS — CHANGES_REQUESTED** [review]

File: be/src/controlers/health.router.ts (hunk 3 above)
Comment (ja-fra) [reply be/src/controlers/health.router.ts]: For a start, response code `200` / `5xx` is sufficient.

In future we could add more details, especially if we start to have more component integrated into the healthcheck.

On the other hand - having a simple payload with a human readable status is convenient in many cases.

So I'd suggest the following payload for the health-response:
```json
{
  "status": "OK", // or "ERROR"
  "details": {
    "mongodb": "OK" // or "ERROR"
  }
}
```

File: be/src/controlers/health.router.ts (hunk 4 above)
Comment (ja-fra) [reply be/src/controlers/health.router.ts]: See [comment above](#discussion_r762227273)

**Review by ja-fra — CHANGES_REQUESTED** [review]
Small cleanups, but overall LGTM :+1:

File: be/tsconfig.json [hunk 8]
```diff
@@ -6,16 +6,20 @@
...
     "baseUrl": "./",
     "types": ["jest"],
     "outDir": "dist/"
   },
+  "include": ["src/**/*.ts"],
   "exclude": [
     "node_modules/",
-    "jest.config.ts"
+    "jest.config.ts",
+    "src/generated/"
```
Comment (ja-fra) [inline be/tsconfig.json:22]: Why should this be excluded? If we generate `.ts`, it needs to be compiled...

File: be/tsconfig.json [hunk 9]
```diff
@@ -6,16 +6,20 @@
...
     "sourceMap": false,
     "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
-    "rootDir": "./",
+    "rootDir": "./src",
     "baseUrl": "./",
     "types": ["jest"],
     "outDir": "dist/"
   },
+  "include": ["src/**/*.ts"],
```
Comment (ja-fra) [inline be/tsconfig.json]: Is this still required? The `rootDir` has been updated to `./src`...

File: be/tsconfig.json (hunk 8 above)
Comment (ManuelTS) [reply be/tsconfig.json:22]: The `jest.config.ts` is just there for configuring jest to run all tests. This file is not needed in the development build

**Review by ManuelTS — APPROVED** [review]

File: be/tsconfig.json (hunk 9 above)
Comment (ManuelTS) [reply be/tsconfig.json]: Has been removed

**Pull request merged**